### PR TITLE
Add tek/splain plugin

### PR DESCRIPTION
**Pull request by megri**

### CODE REVIEW

File: stage2/plugins/SplainPlugin.scala [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  def scalacOptions( jarPath: File, params: Seq[ParameterPair] ) =
+    Seq( "-Xplugin:" ++ jarPath.string ) ++ mkParamList( params )
+
+  private def mkParamList(params: Seq[ParameterPair]) =
+    params.foldLeft( Seq.empty[String] ) {
+      case (paramList, (param, Some(value))) =>
+        paramList :+ s"-P:splain:$param:$value"
+      case (paramList, (param, None)) =>
+        paramList :+ s"-P:splain:$param"
+    }
```
Comment (cvogt) [inline stage2/plugins/SplainPlugin.scala]: .map seems like the more appropriate operator here. and value.map(":"+_).getOrElse("") would case the second case.

File: stage2/plugins/SplainPlugin.scala [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+/**
+  * A scala compiler plugin for more concise errors
+  * See: https://github.com/tek/splain
+  */
+trait SplainPlugin extends BaseBuild{
+  import SplainPlugin.ParameterPair
+
+  def splainVersion = "0.2.4"
+
+  def splainArguments: Seq[ParameterPair] = Seq( "all" -> None )
```
Comment (cvogt) [inline stage2/plugins/SplainPlugin.scala]: How about we make a sealed trait with case objects for the cases?

Comment (megri) [inline stage2/plugins/SplainPlugin.scala]: I actually started down that path but wanted some feedback first :)

File: stage2/plugins/SplainPlugin.scala (hunk 1 above)
Comment (megri) [inline stage2/plugins/SplainPlugin.scala]: Fair enough!

Comment (megri) [inline stage2/plugins/SplainPlugin.scala]: Are you strongly opposed to having two cases? I think it reads a bit cleaner personally

**Comment by megri** [comment]
Please review my changes. 

I tried scoping the members inherited from the trait but they still show up in the list shown when running `cbt` from a project whose build extends the plugin.

**Comment by megri** [comment]
Reading up on access modifiers in Scala, `protected[this]` should make the member accessible only in the inner scope of the extending class. I've created an issue that reflects (heh!) the culprit here: https://github.com/cvogt/cbt/issues/522

**Pull request closed without merging**